### PR TITLE
Use a single symlink that always points to the nodejs version being used

### PR DESCRIPTION
**Pull request by antitoxic**
I find that it's useful:

1. In places you are not in fish shell but you still need to access the nodejs bin dir. Otherwise you need to do some gymnastics with the getting the version file and constructing the path. 
1. Places you need to hardcode a path but you still want to point to the nodejs bin you are using day to day and not update it every time you start using a new version.
1. A bit simpler than cleaning fish_user_paths from previous paths every time you change version

I don't mind this not being merged. Just opening for clarity.

**Edit** : closed primarily due to lack of motivation as @jorgebucaran didn't hint his intentions and was acting agressive

### CODE REVIEW

**Comment by jorgebucaran** [comment]
Hey! @antitoxic Is this a breaking change?

**Comment by antitoxic** [comment]
It's not but because I've removed the fish_user_paths cleanup it will not remove the last bin for from there. I can  add it back for better compatibility. Although it will be confusing to read in the future.

**Comment by jorgebucaran** [comment]
@antitoxic Can you elaborate more? What happens after I merge this? From a regular user's standpoint, what difference does this make / what's this feature about?

File: functions/nvm.fish
Comment (jorgebucaran) [inline functions/nvm.fish:198]: @antitoxic Please use `command ln`.

File: functions/nvm.fish
Comment (jorgebucaran) [inline functions/nvm.fish:201]: @antitoxic What's the purpose of `/current/bin` and what is this `current` prefix that I haven't heard about?

**Review by jorgebucaran — CHANGES_REQUESTED** [review]

**Comment by jorgebucaran** [comment]
@antitoxic Is this a feature or a bug fix -- both?

**Comment by ljharb** [comment]
Note that nvm had, but removed, this feature when it became apparent that implicitly doing this simply *does not work* when the node version can be different across multiple shells.

If i have one tab open on node 12, and one on node 10, which is “current”?

**Comment by antitoxic** [comment]
>  Is this a feature or a bug fix -- both

Feature

> Note that nvm had, but removed, this feature when it became apparent that implicitly doing this simply does not work when the node version can be different across multiple shells.

Ok, i don't mind not being merged. I'm going to use my fork with which I'm satisfied. Running 2 different versions of node at the same time is not a use-case for me.

Thanks.

File: functions/nvm.fish
Comment (antitoxic) [reply functions/nvm.fish:201]: It's not a prefix that should be heard. Just a name. If you prefer it can be `inuse`, `used`, etc. It's a symlink to the currently in use nodejs .

**Comment by jorgebucaran** [comment]
@antitoxic You can use your fork or do whatever you want. You can't expect people to merge a random PR without understanding from the maintainer and discussing things thoroughly. 

Now, fish-nvm does not seek feature parity with the original nvm project. I don't monitor nvm-sh/nvm and have really no clue how it works. I have also never used 2 different Node.js versions at the same time, so I was rather interested in this PR and having this discussion. I'm usually positive about merging changes that simplify code, even if that means breaking backward compatibility or removing features.

File: functions/nvm.fish
Comment (jorgebucaran) [reply functions/nvm.fish:201]: @antitoxic In other words, this is an arbitrary name?

**Comment by antitoxic** [comment]
@jorgebucaran absolutely agree you should understand. That's the level I can explain. You didn't hint you feel favourable of the PR even once, so why should I spend more energy explaining?

That's why I closed it. Feel free to reopen.

Maybe a useful note:

> @antitoxic You can use your fork or do whatever you want. You can't expect people to merge a random PR without understanding from the maintainer and discussing things thoroughly.

This sounds aggressive. No need to. You can also do whatever you want. You can't expect PRs if I don't understand you are thinking of merging or not. Doesn't that sound agressive? Let's have less of this kinda words.

**Comment by jorgebucaran** [comment]
Sounds like you were impatient to merge this, and since that wasn't happening promptly enough, chose to run away instead, but not after issuing your proclamation that you'd use your own fork. Do whatever you want.

**Comment by jorgebucaran** [comment]
@antitoxic Sounds like you were impatient to merge this, and since that wasn't happening promptly enough, chose to run away instead, but not after issuing your proclamation that you'd use your own fork. Do whatever you want.

**Comment by jorgebucaran** [comment]
Sounds like you were impatient to merge this, and since that wasn't happening promptly enough, chose to run away instead, but not after issuing your proclamation that you'd use your own fork. Too bad.

**Pull request closed without merging**
